packages/fx-core/src/core/middleware/projectMigratorV3.ts

```diff
@@ -366,11 +336,80 @@ export async function configsMigration(context: MigrationContext): Promise<void>
               FileType.CONFIG,
               bicepContent
             );
-            await context.fsWriteFile(path.join(SettingsFolderName, ".env." + envName), envData);
+            await context.fsWriteFile(path.join(SettingsFolderName, ".env." + envName), envData, {
+              encoding: "utf8",
+              flag: "a+",
```

Comment:
add comments for the options?

Reply:
good suggestion

Reply:
and maybe mode property should be default, we just need to use flag option {encoding: "utf8", flag: "a+"}